assets/components/src/wizard-pagination/index.js

```diff
@@ -22,38 +16,93 @@ import './style.scss';
 
 const { withRouter } = Router;
 
-class WizardPagination extends Component {
-	/**
-	 * Render.
-	 */
-	render() {
-		const { history, location, routes } = this.props;
-		if ( ! routes || ! history || ! location ) {
-			return;
-		}
-		const currentIndex = parseInt( routes.indexOf( location.pathname ) );
-		if ( 0 === currentIndex ) {
-			return <Fragment />;
+const WizardPagination = props => {
+	const [ showSteps, setShowSteps ] = useState( false );
+	const stepper = useRef( null );
+	const { location, routes } = props;
+	const routeList = Object.keys( routes || {} );
+	const currentRoute = routeList.find( route => location.pathname === routes[ route ].path );
+
+	useEffect( () => {
+		window.addEventListener( 'click', hideSteps );
+		return () => window.removeEventListener( 'click', hideSteps );
+	}, [] );
+
+	useEffect( () => {
+		setShowSteps( false );
+	}, [ currentRoute ] );
+
+	const hideSteps = e => {
+		// If clicking outside the expanded stepper, hide it.
+		if (
+			stepper.current &&
+			e.target !== stepper.current &&
+			! e.target.classList.contains( 'newspack-wizard-pagination__show-steps' )
+		) {
+			setShowSteps( false );
 		}
-		return (
-			<Fragment>
-				<div className="newspack-wizard-pagination">
-					<div className="newspack-wizard-pagination__pagination">
-						{ __( 'Step' ) } { currentIndex } { __( 'of' ) } { routes.length - 1 }{' '}
-					</div>
-					<Button
-						isLink
-						isSmall
-						className="newspack-wizard-pagination__navigation"
-						onClick={ () => history.goBack() }
-					>
-						<ArrowBackIcon />
-						{ __( 'Back' ) }
-					</Button>
-				</div>
-			</Fragment>
-		);
-	}
-}
+	};
+
+	return (
+		<Fragment>
+			<div className="newspack-wizard-pagination newspack-wizard__header__inner">
+				{ routeList.length && (
+					<>
+						<Button
+							className="newspack-wizard-pagination__show-steps"
+							onClick={ () => setShowSteps( ! showSteps ) }
+						>
+							{ routes[ currentRoute ].title }
+						</Button>
+						<ul
+							className={ `newspack-wizard-pagination__steps ${ ! showSteps ? 'hidden' : '' }` }
+							ref={ stepper }
+						>
+							{ routeList.map( ( route, index ) => {
+								if ( 'welcome' === route ) {
+									return null;
+								}
+
+								const currentIndex = routeList.indexOf( currentRoute );
+								const classes = [];
+
+								if ( route === currentRoute ) {
+									classes.push( 'active' );
+								}
+
+								if ( index < currentIndex ) {
+									classes.push( 'complete' );
+								}
+
+								return (
+									<>
+										<li key={ index } className="newspack-wizard-pagination__step">
```

Comment:
The `key` prop should actually be on the `Fragment` which is the parent of this `li`

Reply:
True, fixed in 01aedc7